Expand "Update the connectivity module firmware on UNO R4 WiFi" to cover additional methods (#284)
arduino/help-center-content: content/Hardware Support/UNO/Update-the-connectivity-module-firmware-on-UNO-R4-WiFi.md

id: 9670986058780
---

Learn how to update the firmware of the ESP32-S3 connectivity module on the UNO R4 WiFi.

Updating the firmware is required to use UNO R4 WiFi with Arduino Cloud, and can resolve issues with UNO R4 WiFi not being detected by Arduino IDE and other development tools.

In this article:

* [Use the Firmware Updater in Arduino IDE](#ide)
* [Use IoT Cloud to update the firmware](#iot)
* [Use the updater script](#unor4wifi-updater)
* [Run espflash directly](#espflash)

---

<a id="ide"></a>

## Use the Firmware Updater in Arduino IDE

Updating the connectivity firmware is easy when using the Firmware Updater in Arduino IDE 2.2.1 or later.

> <a class="link-external" href="https://support.arduino.cc/hc/en-us/articles/360013896579-Use-the-Firmware-Updater-in-Arduino-IDE">Use the Firmware Updater in Arduino IDE</a>

Reviewer: ```suggestion
```
I would consider removing this link since there is no other relevant information for UNO R4 WiFi in that article (and the guide in this section is already tailored for the UNO R4 Wifi).

Author: Very true, that was a remnant from before I decided to include the full steps!

    ![Firmware Updater window displaying selected firmware version and the "install" button](img/firmware-updater-arduino-ide-2-select-board-install.png)

6. Wait until the text "Firmware successfully installed" is displayed.

    !["Firmware successfully installed" message displaying on the Firmware Updater window](img/firmware-updater-arduino-ide-2-installation-successful.png)

7. Disconnect and reconnect the UNO R4 WiFi board from your computer.

   > **Warning:** After flashing the firmware on the UNO R4 WiFi, the board will remain in **ESP Download** mode until you disconnect and reconnect it from your computer. If you upload a sketch while the board is in ESP Mode, it will erase the special firmware that lets the ESP32 chip function as an USB bridge.

8. Close the Firmware Updater by clicking the "x" in the top-right corner of the Firmware Updater window.

    ![Closing the Firmware Updater window.](img/firmware-updater-arduino-ide-2-close.png)

---

<a id="iot"></a>

## Use IoT Cloud to update the firmware

When you add a new device to IoT Cloud, the connectivity module firmware is automatically updated.

> <a class="link-external" href="https://support.arduino.cc/hc/en-us/articles/10501616961564-Update-connectivity-module-firmware-with-IoT-Cloud">Learn more</a>

Reviewer: ```suggestion
<a class="link-chevron-right" href="https://support.arduino.cc/hc/en-us/articles/10501616961564-Update-connectivity-module-firmware-with-IoT-Cloud">Learn more</a>
```

1. Close the terminal window.
1. Disconnect the USB cable of the **UNO R4 WiFi** board from your computer.
1. Connect the **UNO R4 WiFi** board to your computer with the USB cable again.

---

<a id="espflash"></a>

## Run espflash directly

> The [updater script](#unor4wifi-updater) will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).

Reviewer: ```suggestion
The [Firmware Updater in Arduino IDE](#ide) and the [updater script](#unor4wifi-updater) will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).
```
The [Firmware Updater in Arduino IDE](#ide) will also not work if the board is not identified as UNO R4 (as written in the note in the [Use the Firmware Updater in Arduino IDE](#ide) section)

Author: That's a good point, but we might as well expand it to also cover the IoT Cloud method:

```suggestion
The above methods will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).
```
